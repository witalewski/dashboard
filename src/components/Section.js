import React from 'react';
import styled from '@emotion/styled';
import { borderColor, sectionMargin } from '../global/styleConstants';

const SectionStyled = styled.section`
  margin: ${sectionMargin}px;
  padding: 32px;

  background: white;
  border: 1px solid ${borderColor};
  border-radius: 5px;

  font-size: 14px;

  .title {
    font-size: 16px;
    text-transform: uppercase;
  }
`;

export const Section = ({ className, title, children }) => (
  <SectionStyled className={className}>
    <h2 className="title">{title}</h2>
    {children}
  </SectionStyled>
);
