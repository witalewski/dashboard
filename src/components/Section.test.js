import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { Section } from './Section';

describe('Section', () => {
  it('matches snapshot', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Section title="Dashboard" />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
