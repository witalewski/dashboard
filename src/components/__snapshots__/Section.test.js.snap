// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section matches snapshot 1`] = `
<ForwardRef(render)
  className="section"
>
  <h2
    className="title"
  >
    Dashboard
  </h2>
  <div
    className="content"
  >
    Section content
  </div>
</ForwardRef(render)>
`;
